packages/lodestar/src/sync/regular/naive/naive.ts

```diff
@@ -154,12 +169,43 @@ export class NaiveRegularSync implements IRegularSync {
   }
 
   private getSyncPeers = async (): Promise<PeerId[]> => {
-    return this.network.getPeers().reduce( (validPeers: PeerId[], peer: PeerId) => {
-      const rep = this.reps.getFromPeerId(peer);
-      if(rep && rep.latestStatus) {
-        validPeers.push(peer);
+    if (!this.network.getPeers().includes(this.bestPeer)) {
+      // bestPeer disconnected
+      this.bestPeer = undefined;
+      await this.waitForBestPeer();
+    }
+    return [this.bestPeer];
+  };
+
+  private waitForBestPeer = async (): Promise<void> => {
+    // check every slot
+    const waitingTime = this.config.params.SECONDS_PER_SLOT * 1000;
+    const state = await this.chain.getHeadState();
+    this.controller = new AbortController();
+    let isAborted = false;
+    this.controller.signal.addEventListener("abort", () => {
+      this.logger.verbose("RegularSync: Abort waitForBestPeer");
+      isAborted = true;
+    });
+    while (!this.bestPeer && !isAborted) {
+      // wait first to make sure we have latest status
+      await sleep(waitingTime);
+      const status = createStatus(this.chain);
+      const previousSlot = getCurrentSlot(this.config, state.genesisTime) - 1;
+      await syncPeersStatus(this.reps, this.network, status);
```

Comment:
I would separate process of obtaining status so it's running during both initial and regular sync on some time interval